### ISSUE TITLE
dbgenerated function cause Null constraint error in MySQL

### ISSUE THREAD

**Issue by qhnu** [prisma/prisma#12850]
### Bug description

Some dbgenerated functions work and others produce errors.
The error is not on the database side, but on Prisma Client side.

### How to reproduce

✅ no error:
```prisma
model A {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid()))")) @db.Binary(16)
}
```

❌ error:
```prisma
model B {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid(), 1))")) @db.Binary(16)
}
```

error detail:
```
Null constraint violation on the (not available)
  code: 'P2011',
  clientVersion: '3.12.0',
  meta: { constraint: null }
```

### Expected behavior

Because the value is generated on the db side, there is no need to detect errors in Prisma Client.

### Prisma information

```prisma
datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity", "fullTextIndex", "fullTextSearch", "interactiveTransactions", "filterJson"]
}

model A {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid()))")) @db.Binary(16)
}

model B {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid(), 1))")) @db.Binary(16)
}
```

```ts
export const test = async () => {
  await prisma.a.create({ data: {} })
  await prisma.b.create({ data: {} })
}
```

### Environment & setup

- OS: Mac OS
- Database: MySQL
- Node.js version: v16.14.0

### Prisma Version

```
prisma                  : 3.12.0
@prisma/client          : 3.12.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 22b822189f46ef0dc5c5b503368d1bee01213980 (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : 22b822189f46ef0dc5c5b503368d1bee01213980
Studio                  : 0.459.0
Preview Features        : referentialIntegrity, fullTextIndex, fullTextSearch, interactiveTransactions, filterJson
```

**Comment by janpio** [comment]
I think the error message actually comes from the database and bubbles up. Can you manually add data to the table with SQL? 

Did you get this schema by introspecting an existing database (hence the SQL is definitely correct), or did you write this schema and then use Prisma to migrate it (which might make it useful to check if the underlying SQL DDL is indeed correct)?

Can you log the queries that are executed by prisma with https://pris.ly/d/logging and https://pris.ly/d/debugging and share them please?

**Comment by qhnu** [comment]
Thank you for your reply.

> Can you manually add data to the table with SQL? Did you get this schema by introspecting an existing database

Yes, I generated `*.prisma` file by `prisma db pull` from a database running with manual input.

Introspection generated `*.prisma`:
```prisma
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model A {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid()))")) @db.Binary(16)
}

model B {
  id Bytes @id @default(dbgenerated("(uuid_to_bin(uuid(),1))")) @db.Binary(16)
}
```

> I think the error message actually comes from the database and bubbles up. 
> Can you log the queries that are executed by prisma

That was right.
According to the detailed logs, the cause of the null error appears to be the absence of `SELECT uuid_to_bin...`.

```log
prisma:info Starting a mysql pool with 1 connections.
prisma:query BEGIN
prisma:query SELECT uuid_to_bin(uuid()) AS `id`
prisma:query INSERT INTO `ok`.`A` (`id`) VALUES (?)
prisma:query SELECT `ok`.`A`.`id` FROM `ok`.`A` WHERE `ok`.`A`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT

prisma:query BEGIN
prisma:query INSERT INTO `ok`.`B` (`id`) VALUES (?)
prisma:query ROLLBACK
PrismaClientKnownRequestError: 
Invalid `prisma.b.create()` invocation in
/Users/admin/Git/lunch/src/libs/fetch.ts:5:18

  2 
  3 export const test = async () => {
  4   await prisma.a.create({ data: {} })
→ 5   await prisma.b.create(
  Null constraint violation on the (not available)
    at Object.request (/Users/admin/Git/lunch/node_modules/@prisma/client/runtime/index.js:45578:15)
    at async PrismaClient._request (/Users/admin/Git/lunch/node_modules/@prisma/client/runtime/index.js:46405:18)
    at Object.test (/Users/admin/Git/lunch/src/libs/fetch.ts:5:3) {
  code: 'P2011',
  clientVersion: '3.12.0',
  meta: { constraint: null }
}
```

Perhaps there are no ```SELECT uuid_to_bin(uuid(),1) AS `id\`;``` commands that need to be issued.
It has been confirmed that ```SELECT uuid_to_bin(uuid(),1) AS `id\`;``` commands return normal values when manually entered from database.

> use Prisma to migrate it

Since this is `planetscale` environment, the migrate command is not used.

**Comment by janpio** [comment]
> Perhaps there are no `` SELECT uuid_to_bin(uuid(),1) AS `id`; `` commands that need to be issued.
> It has been confirmed that `` SELECT uuid_to_bin(uuid(),1) AS `id`; `` commands return normal values when manually entered from database.

That is _exactly_ the problem here - thanks for the additional data!

As you can see we have an additional query `SELECT uuid_to_bin(uuid()) AS id` to have the ID available (which MySQL unfortunately does not return when it is generated during the query itself - making it impossible for us to read back the full inserted row. It seems we are missing the special handling for the usage of [`swap_flag`](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-to-bin) (which I did not know existed until today!).

**Comment by janpio** [comment]
This should be fixed in the `@dev` version of Prisma. Can you try installing it temporarily in your app and see if it now works? Thanks.

**Comment by qhnu** [comment]
Thank you for your quick response.
We have confirmed that the following version works correctly.
```json
  "dependencies": {
    "@prisma/client": "3.15.0-dev.20",
    "prisma": "3.15.0-dev.20"
  }
```
```
prisma:info Starting a mysql pool with 9 connections.
prisma:query BEGIN
prisma:query SELECT uuid_to_bin(uuid()) AS `id`
prisma:query INSERT INTO `ok`.`A` (`id`) VALUES (?)
prisma:query SELECT `ok`.`A`.`id` FROM `ok`.`A` WHERE `ok`.`A`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
prisma:query BEGIN
prisma:query SELECT uuid_to_bin(uuid(), 1) AS `id`
prisma:query INSERT INTO `ok`.`B` (`id`) VALUES (?)
prisma:query SELECT `ok`.`B`.`id` FROM `ok`.`B` WHERE `ok`.`B`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
```
No errors.
Thank you!

**Comment by sidvenu** [comment]
@janpio any idea when this is coming to prisma v4?

**Comment by janpio** [comment]
We are not aware of that. For us this was solved and fixed in 3.15.0, and we haven't heard otherwise yet.
Please open a new issue and describe exactly the problem you are experiencing, and we can look into that.

**Comment by sidvenu** [comment]
@janpio done - #15727
